core/kt-fast-collections-generator/src/main/kotlin/collections/ArrayList.kt

```diff
@@ -250,6 +250,16 @@ private fun CollectionItemType.generateArrayList(context: GeneratorContext, curr
                 res.add(c)
                 return res
             }
+
+            /** GENERATED CODE */
+            fun ${paramsDecl} mutable${simpleName}ArrayListOf(a: $type, b: $type, c: $type, d: $type): Mutable${simpleName}ArrayList${paramsUse} {
```

Comment:
Might be possible to use `vararg` here. Would remove the necessity to reimplement this method every time we want to add another argument.

Reply:
I opened an issue for that: #6337